src/fraposa_pgsc/fraposa.py

```diff
@@ -136,13 +138,16 @@ def read_bed(bed_filepref, dtype=np.int8, filt_iid=None):
     p = len(bim)
     n = len(fam)
 
-    if type(filt_iid) is list:
-        matched_ids = set(filt_iid).intersection(fam['iid'])
+    if filt_iid:
+        fam_ids = set(SampleID(x, y) for x, y in zip(fam['fid'], fam['iid'], strict=True))
+        matched_ids = filt_iid.intersection(fam_ids)
         if len(matched_ids) == 0:
-            logging.error('ERROR: 0 / {} ids in filter list match the study dataset'.format(len(filt_iid)))
-            sys.exit(1)
+            raise ValueError(f"ERROR: 0 / {len(filt_iid)} ids in filter list match the study dataset")
+        elif len(fam_ids) != len(fam):
+            raise ValueError("Samples with duplicated FID + IID detected, please remove and retry")
+
         bed = np.zeros(shape=(p, len(matched_ids)), dtype=dtype)
-        fam_mask = fam.iid.isin(matched_ids)
+        fam_mask = fam.iid.isin((x.IID for x in matched_ids)) & fam.fid.isin(x.FID for x in matched_ids)
```

Comment:
I think this might fail in weird ways, shouldn't it check in ['F1','I1'] is in [['F1','I2'], ['F1','I3'], ['F2','I3']]. The current criteria would include this ID even though F1/I1 never occur together in the list.